src/reader.js

```diff
@@ -148,8 +148,13 @@ class Reader extends EventEmitter {
 
     // Trigger a query of the configured `lookupdHTTPAddresses`.
     const endpoint = this.roundrobinLookupd.next();
-    lookup(endpoint, this.topic, (err, nodes = []) =>
-      nodes.map(n => this.connectToNSQD(n.broadcast_address, n.tcp_port)));
+    console.log(endpoint);
+    lookup(endpoint, this.topic, (err, nodes = []) => {
+      if(err){
```

Comment:
I think it was somewhat by design previously to swallow lookup errors. Technically, the Reader shouldn't die just because a lookupd has an issue returning a response. That would be bad for a Reader happily connected to a nsqds. That said, not ideal in terms of debugging a lookupd issue, but it might warrant another solution.